### PR TITLE
feat(provider): unsplash

### LINKED ISSUE #353: feat: unsplash support
https://unsplash.com/developers

Can you add support for this?

### PR DESCRIPTION
**Pull request by shadow81627**
resolves #353

### CODE REVIEW

File: docs/content/en/4.providers/glide.md [hunk 1]
```diff
@@ -2,10 +2,10 @@
 title: Glide Provider
 description: 'Nuxt Image has first class integration with Glide'
 navigation:
-  title: glide
+  title: Glide
```
Comment (shadow81627) [inline docs/content/en/4.providers/glide.md:5]: I capitalised the Glide title in the docs since it was annoying me that I made it lowercase.

**Comment by shadow81627** [comment]
I didn't realise before I created this provider that there is an alias feature that is used by ipx for unsplash in the playground. Could I get some help with improving the unsplash provider docs to mention ipx unsplash alias support?

File: test/unit/providers.test.ts [hunk 2]
```diff
@@ -135,6 +136,18 @@ describe('Providers', () => {
...
 
+  test('unsplash', () => {
+    const providerOptions = {
+      baseURL: ''
+    }
+
+    for (const image of images) {
+      const [src, modifiers] = image.args
+      const generated = imgix.getImage(src, { modifiers, ...providerOptions }, emptyContext)
+      expect(generated).toMatchObject(image.imgix)
```
Comment (pi0) [inline test/unit/providers.test.ts:147]:
```suggestion
      expect(generated).toMatchObject(image.unsplash)
```

File: test/unit/providers.test.ts [hunk 3]
```diff
@@ -135,6 +136,18 @@ describe('Providers', () => {
...
   })
 
+  test('unsplash', () => {
+    const providerOptions = {
+      baseURL: ''
+    }
+
+    for (const image of images) {
+      const [src, modifiers] = image.args
+      const generated = imgix.getImage(src, { modifiers, ...providerOptions }, emptyContext)
```
Comment (pi0) [inline test/unit/providers.test.ts:146]:
```suggestion
      const generated = unsplash.getImage(src, { modifiers, ...providerOptions }, emptyContext)
```

File: docs/content/en/4.providers/unsplash.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+---
+title: Unsplash Provider
+description: 'Nuxt Image has first class integration with Unsplash'
+navigation:
+  title: Unsplash
+---
+
+Integration between [Unsplash](https://unsplash.com/documentation#dynamically-resizable-images) and the image module.
```
Comment (pi0) [inline docs/content/en/4.providers/unsplash.md:8]: Maybe it is worth mentioning about [License](https://unsplash.com/license)? (mostly in good faith of using the service)

File: test/providers.ts [hunk 5]
```diff
@@ -7,6 +7,7 @@ export const images = [
     fastly: { url: '/test.png' },
     glide: { url: '/test.png' },
     imgix: { url: '/test.png' },
+    unsplash: { url: '/unsplash/photo-1606112219348-204d7d8b94ee' },
```
Comment (pi0) [inline test/providers.ts:10]: Why we need `/unsplash`? Correct URL seems like [this](https://images.unsplash.com/photo-1606112219348-204d7d8b94ee?w=200) (notice the one I used in current examples was for showcasing alias)

**Comment by pi0** [comment]
> I didn't realise before I created this provider that there is an alias feature that is used by ipx for unsplash in the playground. Could I get some help with improving the unsplash provider docs to mention ipx unsplash alias support?

Sure. Feel free changing `alias` example to something else :)

File: test/providers.ts (hunk 5 above)
Comment (shadow81627) [reply test/providers.ts:10]: Thanks I was a little confused by the alias.

```suggestion
    unsplash: { url: '/photo-1606112219348-204d7d8b94ee' },
```

File: test/providers.ts [hunk 6]
```diff
@@ -20,6 +21,7 @@ export const images = [
     fastly: { url: '/test.png?width=200' },
     glide: { url: '/test.png?w=200' },
     imgix: { url: '/test.png?w=200' },
+    unsplash: { url: '/unsplash/photo-1606112219348-204d7d8b94ee?w=200' },
```
Comment (shadow81627) [inline test/providers.ts:24]:
```suggestion
    unsplash: { url: '/photo-1606112219348-204d7d8b94ee?w=200' },
```

File: test/providers.ts [hunk 7]
```diff
@@ -33,6 +35,7 @@ export const images = [
     fastly: { url: '/test.png?height=200' },
     glide: { url: '/test.png?h=200' },
     imgix: { url: '/test.png?h=200' },
+    unsplash: { url: '/unsplash/photo-1606112219348-204d7d8b94ee?h=200' },
```
Comment (shadow81627) [inline test/providers.ts:38]:
```suggestion
    unsplash: { url: '/photo-1606112219348-204d7d8b94ee?h=200' },
```

File: test/providers.ts [hunk 8]
```diff
@@ -46,6 +49,7 @@ export const images = [
     fastly: { url: '/test.png?width=200&height=200' },
     glide: { url: '/test.png?w=200&h=200' },
     imgix: { url: '/test.png?w=200&h=200' },
+    unsplash: { url: '/unsplash/photo-1606112219348-204d7d8b94ee?w=200&h=200' },
```
Comment (shadow81627) [inline test/providers.ts:52]:
```suggestion
    unsplash: { url: '/photo-1606112219348-204d7d8b94ee?w=200&h=200' },
```

File: test/providers.ts [hunk 9]
```diff
@@ -59,6 +63,7 @@ export const images = [
     fastly: { url: '/test.png?width=200&height=200&fit=bounds' },
     glide: { url: '/test.png?w=200&h=200&fit=contain' },
     imgix: { url: '/test.png?w=200&h=200&fit=fill' },
+    unsplash: { url: '/unsplash/photo-1606112219348-204d7d8b94ee?w=200&h=200&fit=fill' },
```
Comment (shadow81627) [inline test/providers.ts:66]:
```suggestion
    unsplash: { url: '/photo-1606112219348-204d7d8b94ee?w=200&h=200&fit=fill' },
```

File: test/providers.ts [hunk 10]
```diff
@@ -72,6 +77,7 @@ export const images = [
     fastly: { url: '/test.png?width=200&height=200&fit=bounds&format=jpeg' },
     glide: { url: '/test.png?w=200&h=200&fit=contain&fm=jpeg' },
     imgix: { url: '/test.png?w=200&h=200&fit=fill&fm=jpeg' },
+    unsplash: { url: '/unsplash/photo-1606112219348-204d7d8b94ee?w=200&h=200&fit=fill&fm=jpeg' },
```
Comment (shadow81627) [inline test/providers.ts:80]:
```suggestion
    unsplash: { url: '/photo-1606112219348-204d7d8b94ee?w=200&h=200&fit=fill&fm=jpeg' },
```

File: playground/providers.ts [hunk 11]
```diff
@@ -296,6 +296,23 @@ export const providers: Provider[] = [
       }
     ]
   },
+  // Unsplash
+  {
+    name: 'unsplash',
+    samples: [
+      {
+        src: '/unsplash/photo-1606112219348-204d7d8b94ee',
```
Comment (shadow81627) [inline playground/providers.ts:304]:
```suggestion
        src: '/photo-1606112219348-204d7d8b94ee',
```

File: docs/content/en/4.providers/unsplash.md (hunk 4 above)
Comment (shadow81627) [reply docs/content/en/4.providers/unsplash.md:8]:
```suggestion

Integration between [Unsplash](https://unsplash.com/documentation#dynamically-resizable-images) and the image module. See [Unsplash License](Unsplash photos are made to be used freely.) for what usage is permitted.
```

File: src/runtime/providers/unsplash.ts [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+import { joinURL } from 'ufo'
+import type { ProviderGetImage } from 'src'
+import { operationsGenerator } from './imgix'
+
+const unsplashCDN = 'https://images.unsplash.com/'
+
+export const getImage: ProviderGetImage = (src, { modifiers = {}, baseURL = unsplashCDN } = {}) => {
+  const operations = operationsGenerator(modifiers)
+  return {
+    url: joinURL(src.startsWith(baseURL) ? '' : baseURL, src + (operations ? ('?' + operations) : ''))
```
Comment (pi0) [inline src/runtime/providers/unsplash.ts:12]: Tip: we can use withBase utility from ufo that ensures base without duplicate

**Review by pi0 — APPROVED** [review]

File: docs/content/en/4.providers/glide.md [hunk 13]
```diff
@@ -2,18 +2,18 @@
...
+Integration between [Glide](https://glide.thephpleague.com/) and the image module.
 
 To use this provider you just need to specify the base url of your service in glide.
 
 ```js{}[nuxt.config.js]
 export default {
   image: {
     glide: {
-      baseURL: 'https://glide.example.com'
+      baseURL: 'https://glide.herokuapp.com/1.0/'
```
Comment (pi0) [inline docs/content/en/4.providers/glide.md:16]: Note for later improvement: We probably need better documentation that this `baseURL` is a laravel application of users not a fixed base to example.

Comment (shadow81627) [reply docs/content/en/4.providers/glide.md:16]:
```suggestion
      // baseURL of your laravel application
      baseURL: 'https://glide.herokuapp.com/1.0/'
```

**Pull request merged**